src/views/ApartmentApp/components/StudentApplication/index.jsx

```diff
@@ -561,6 +572,11 @@ const StudentApplication = ({ userProfile }) => {
   const deleteApartmentApplication = async () => {
     setDeleting(true);
     setDeleteButtonAlertTimeout(null);
+    if (!applicationExists) {
+      setApplicationDetails(loadApplication);
+      setDeleting(false);
+      return;
+    }
```

Comment:
it would be beneficial for readers to have a comment explaining why this condition is needed and why you need to short the function.